Updating `upload.yml` to run Julia tests (#340)
XanaduAI/MrMustard: doc/introduction/basic_reference.md

### The math module
The math module is the backbone of Mr Mustard, which consists in the [Math](https://github.com/XanaduAI/MrMustard/blob/main/mrmustard/math/math_interface.py) interface
Mr Mustard comes with a plug-and-play backends through a math interface. You can use it as a drop-in replacement for tensorflow or numpy and your code will be plug-and-play too!
Here's an example where the ``numpy`` backend is used.

```python
import mrmustard.math as math

math.cos(0.1)  # numpy
```

In a different session, we can change the backend to ``tensorflow``.
```python
import mrmustard.math as math
math.change_backend("tensorflow")

math.cos(0.1)  # tensorflow
```

### Optimization

> The `Optimizer` available in `mrmustard.training` uses Adam underneath the hood for Euclidean parameters and a custom symplectic optimizer for Gaussian gates and states, an unitary optimizer for interferometers, and an orthogonal optimizer for real interferometers.

Reviewer: ```suggestion
The `mrmustard.training.Optimizer` uses Adam underneath the hood for the optimization of Euclidean parameters, a custom symplectic optimizer for Gaussian gates and states and a unitary/orthogonal optimizer for interferometers.
```
